staging/src/k8s.io/apiserver/pkg/storage/etcd3/lease_manager.go

```diff
@@ -23,24 +23,26 @@ import (
 
 	"go.etcd.io/etcd/clientv3"
 	"k8s.io/apiserver/pkg/storage/etcd3/metrics"
-	"k8s.io/klog/v2"
 )
 
 const (
 	defaultLeaseReuseDurationSeconds = 60
-	largeLeaseThreshold              = 5000
+	defaultLeaseMaxObjectCount       = 5000
```

Comment:
I think we should use smallest value that gives clear benefit from batching, as it introduces 
smallest delay to following transactions. 

Based on the data from your experiments it seems that ~900 is pretty good, but probably picking the 'round' 1000 is preferred.